### PR TITLE
Properly cast vm object

### LINKED ISSUE #128: Unable to run "main" branch with JDK 21: Caused by: java.lang.NoSuchMethodException: java.lang.Class.detach()
Anyone else seeing this?
```
Exception seen:
cmdline = /[...]/bin/java --add-exports jdk.jconsole/sun.tools.jconsole=ALL-UNNAMED -jar jmxterm-1.0.5-uber.jar -l 62126 -v silent -i /tmp/jmxterm.tmp
Exception in thread "main" java.lang.RuntimeException: Can't cast sun.tools.attach.AttachProviderImpl@7f3b84b8: 62126 to VirtualMachineDescriptor
	at org.cyclopsgroup.jmxterm.jdk9.Jdk9JavaProcess.startManagementAgent(Jdk9JavaProcess.java:55)
	at org.cyclopsgroup.jmxterm.SyntaxUtils.getUrl(SyntaxUtils.java:46)
	at org.cyclopsgroup.jmxterm.boot.CliMain.execute(CliMain.java:135)
	at org.cyclopsgroup.jmxterm.boot.CliMain.main(CliMain.java:41)
Caused by: java.lang.NoSuchMethodException: java.lang.Class.detach()
	at java.base/java.lang.Class.getMethod(Class.java:2395)
	at org.cyclopsgroup.jmxterm.utils.WeakCastUtils.cast(WeakCastUtils.java:43)
	at org.cyclopsgroup.jmxterm.utils.WeakCastUtils.cast(WeakCastUtils.java:116)
	at org.cyclopsgroup.jmxterm.utils.WeakCastUtils.cast(WeakCastUtils.java:81)
	at org.cyclopsgroup.jmxterm.jdk9.Jdk9JavaProcess.startManagementAgent(Jdk9JavaProcess.java:52)
	... 3 more
```

### PR DESCRIPTION
**Pull request by nyg**
In my previous PR #113 I forgot to pass the vm object to the cast method as noted by @tachun-shen-appier (thanks!). This resulted in the error described in issue #128.

Fixes #128.

### CODE REVIEW

**Comment by nyg** [comment]
Hello @jiaqi, could you please review and merge? Many thanks!

**Review by jiaqi — APPROVED** [review]
